source/Plugins.h

```diff
@@ -22,6 +22,22 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 
 
 
+struct PluginDependencies
+{
+	// Checks if there are any dependencies of any kind.
+	bool IsEmpty() const;
+	// Checks if there are any duplicate dependencies. E.g. the same dependency in both required and conflicted.
+	bool IsValid() const;
+	// The plugins, if any, which are required by this plugin.
+	Set<std::string> required;
```

Comment:
`Set` is actually a map, so it's a bit weird that you use it here as a set which I mean it works but it's weird. Just use `std::set` :smile: